servers/rendering/renderer_rd/shader_rd.h

```diff
@@ -201,20 +213,38 @@ class ShaderRD {
 	// Enable/disable variants for things that you know won't be used at engine initialization time .
 	void set_variant_enabled(int p_variant, bool p_enabled);
 	bool is_variant_enabled(int p_variant) const;
+	int64_t get_variant_count() const;
+	int get_variant_to_group(int p_variant) const;
 
 	// Enable/disable groups for things that might be enabled at run time.
 	void enable_group(int p_group);
 	bool is_group_enabled(int p_group) const;
+	int64_t get_group_count() const;
+	const LocalVector<int> &get_group_to_variants(int p_group) const;
+
+	const String &get_name() const;
 
-	static void set_shader_cache_dir(const String &p_dir);
+	static void shaders_embedded_set_lock();
+	static const ShaderVersionPairSet &shaders_embedded_set_get();
+	static void shaders_embedded_set_unlock();
+
+	static void set_shader_cache_user_dir(const String &p_dir);
+	static const String &get_shader_cache_user_dir();
+	static void set_shader_cache_res_dir(const String &p_dir);
+	static const String &get_shader_cache_res_dir();
```

Comment:
The ``const String &p_dir`` has to be changed to ``const String p_dir``.  Same for the ``cpp`` implementation in ``shader_rd.cpp``.  When debugging on Android, the value of the underlying String changes to ``null`` as the game runs.  I'm not sure why this is as I'm unfamiliar with how ``String`` operates, but I confirmed in my debugger here:

https://github.com/godotengine/godot/pull/102552#issuecomment-2895958004

Reply:
I suspect there may be some automatic memory management on ``String`` that frees it when the original variable it was assigned to goes out of scope.  This is just a guess though.

Reply:
This doesn't sound right to me. `const String &` is used in a ton of places in the engine and shader_cache_user_dir and res_dir are regular String. It should be copying the cowdata and increasing the ref count.

Reply:
@TCROC Could you perhaps try to breakpoint into the set functions and see if they're called more than once for each of them during the program runtime?

Reply:
Sure thing!  I'll pull in @DarioSamo's latest changes and grab this info for both of you.

Reply:
This issue appears to have gone away.  I'm willing to chalk this up to a bad build cache or something while I was jumping around between branches.  We can mark this as resolved.  If it resurfaces, I'll open a separate issue.

Reply:
Thanks for confirming!